plugins/catalog-react/src/components/EntityOwnerPicker/useFacetsEntities.ts

```diff
@@ -41,19 +41,37 @@ type FacetsInitialRequest = {
  * hook, which is also used by EntityOwnerPicker.
  * In this mode, the EntityOwnerPicker won't show detailed information of the owners.
  */
-export function useFacetsEntities({ enabled }: { enabled: boolean }) {
+export function useFacetsEntities({
+  enabled,
+  selectedEntityKind,
+}: {
+  enabled: boolean;
+  selectedEntityKind?: string;
+}) {
   const catalogApi = useApi(catalogApiRef);
-
-  const [facetsPromise] = useState(async () => {
+  const facetsPromise = useMemo<Promise<Entity[]>>(() => {
```

Comment:
```suggestion
  const facetsPromise = useMemo<Promise<Entity[]>>(async () => {
```

and then you can greatly simplify everything below